src/components/ModalWithForm.js

```diff
@@ -6,26 +6,41 @@ class ModalWithForm extends Modal {
     this._modalForm = this._modalElement.querySelector(".modal__form");
     this._handleFormSubmit = handleFormSubmit.bind(this);
     this._modalInput = this._modalForm.querySelector(".modal__input");
+    this._modalButton = this._modalElement.querySelector(".modal__button");
+  }
+
+  setButtonText(loading) {
+    if (loading) {
+      console.log("saving");
+      this._modalButton.textContent = "SAVING...";
+    }
+    if (loading === false) {
+      this._modalButton.textContent = "SAVE";
+    }
   }
 
   close() {
-    this._modalForm.reset();
     super.close();
   }
 
+  _clearForm() {
+    this._modalForm.reset();
+  }
+
   _getInputValues() {
-    const inputList = [...this._modalForm.querySelectorAll('input')];
+    const inputList = [...this._modalForm.querySelectorAll("input")];
     const inputValues = {};
     for (const input of inputList) {
       inputValues[input.name] = input.value;
-    };
+    }
     return inputValues;
   }
 
   setEventListeners() {
     this._modalForm.addEventListener("submit", (e) => {
       e.preventDefault();
       this._handleFormSubmit(this._getInputValues());
+      this._clearForm();
       this.close();
```

Comment:
* You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well

Reply:
* You haven't corrected my remark https://github.com/DMyers5280/se_project_aroundtheus/pull/1#discussion_r1750987693 


Please, delete `this.close();` from line 39 in `ModalWithForm`